src/Microsoft.Diagnostics.Monitoring.RestServer/MetricsService.cs

```diff
@@ -20,53 +22,86 @@ internal sealed class MetricsService : BackgroundService
-            _options = metricsOptions.Value;
+            _optionsMonitor = optionsMonitor;
         }
         
-        protected override Task ExecuteAsync(CancellationToken stoppingToken)
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
-            return Task.Run( async () =>
+            while (!stoppingToken.IsCancellationRequested)
             {
-                while (!stoppingToken.IsCancellationRequested)
+                stoppingToken.ThrowIfCancellationRequested();
+
+                try
                 {
-                    stoppingToken.ThrowIfCancellationRequested();
+                    //TODO In multi-process scenarios, how do we decide which process to choose?
+                    //One possibility is to enable metrics after a request to begin polling for metrics
+                    IProcessInfo pi = await _services.GetProcessAsync(filter: null, stoppingToken);
 
-                    try
-                    {
-                        //TODO In multi-process scenarios, how do we decide which process to choose?
-                        //One possibility is to enable metrics after a request to begin polling for metrics
-                        IProcessInfo pi = await _services.GetProcessAsync(filter: null, stoppingToken);
+                    var client = new DiagnosticsClient(pi.EndpointInfo.Endpoint);
 
-                        var client = new DiagnosticsClient(pi.EndpointInfo.Endpoint);
+                    MetricsOptions options = _optionsMonitor.CurrentValue;
+                    using var optionsTokenSource = new CancellationTokenSource();
 
-                        _counterPipeline = new EventCounterPipeline(client, new EventPipeCounterPipelineSettings
-                        {
-                            CounterGroups = Array.Empty<EventPipeCounterGroup>(),
-                            Duration = Timeout.InfiniteTimeSpan,
-                            RefreshInterval = TimeSpan.FromSeconds(_options.UpdateIntervalSeconds)
-                        }, metricsLogger: new[] { new MetricsLogger(_store.MetricsStore) });
+                    //If metric options change, we need to cancel the existing metrics pipeline and restart with the new settings.
+                    using IDisposable monitorListener = _optionsMonitor.OnChange((_, _) => optionsTokenSource.Cancel());
 
-                        await _counterPipeline.RunAsync(stoppingToken);
-                    }
-                    catch (Exception e) when (!(e is OperationCanceledException))
+                    EventPipeCounterGroup[] counterGroups = Array.Empty<EventPipeCounterGroup>();
+                    if (options.Providers.Count > 0)
                     {
-                        //Most likely we failed to resolve the pid. Attempt to do this again.
-                        if (_counterPipeline != null)
+                        //In the dotnet-monitor case, custom metrics are additive to the default counters.
```

Comment:
If they are additive, how does someone exclude the default counters.

Reply:
Initially I didn't really consider this a scenario, but I will add another setting per your suggestion